packages/by-name/microsoft/tardev-snapshotter/package.nix

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 Edgeless Systems GmbH
+# SPDX-License-Identifier: AGPL-3.0-only
+
+{
+  lib,
+  rustPlatform,
+  microsoft,
+  cmake,
+  protobuf,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "tardev-snapshotter";
+  inherit (microsoft.kata-runtime) version src;
+
+  sourceRoot = "${src.name}/src/tardev-snapshotter";
+
+  cargoHash = "sha256-0Jj0+Dsm2wykNopy1Oga0wSQEGIl66AppwmWfjm14mg=";
+
+  nativeBuildInputs = [
+    cmake
+    protobuf
```

Comment:
I don't think this is necessary? Upstream doesn't seem to use it: https://github.com/search?q=repo%3Amicrosoft%2Fkata-containers+path%3A%2F%5Esrc%5C%2Ftardev-snapshotter%5C%2Fsrc%5C%2F%2F+protobuf&type=code

Reply:
cmake:
```
       >   running: cd "/build/source/src/tardev-snapshotter/target/x86_64-unknown-linux-gnu/release/build/libz-ng-sys-8e96c42a70f8ae49/out/build" && CMAKE_PREFIX_PATH="" "cmake" "/build/tardev-snapshotter-3.2.0.azl1-vendor.tar.gz/libz-ng-sys/src/zlib-ng" "-DBUILD_SHARED_LIBS=OFF" "-DZLIB_COMPAT=OFF" "-DZLIB_ENABLE_TESTS=OFF" "-DWITH_GZFILEOP=ON" "-DCMAKE_INSTALL_PREFIX=/build/source/src/tardev-snapshotter/target/x86_64-unknown-linux-gnu/release/build/libz-ng-sys-8e96c42a70f8ae49/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/cc" "-DCMAKE_BUILD_TYPE=Release"
       >
       >   --- stderr
       >   thread 'main' panicked at /build/tardev-snapshotter-3.2.0.azl1-vendor.tar.gz/cmake/src/lib.rs:1098:5:
       >
       >   failed to execute command: No such file or directory (os error 2)
       >   is `cmake` not installed?
```
protobuf:
```
       > error: failed to run custom build command for `containerd-snapshots v0.3.0`
       >
       > Caused by:
       >   process didn't exit successfully: `/build/source/src/tardev-snapshotter/target/release/build/containerd-snapshots-e53e949f2831ba7a/build-script-build` (exit status: 101)
       >   --- stdout
       >   cargo:rerun-if-changed=vendor/github.com/containerd/containerd/api/types/mount.proto
       >   cargo:rerun-if-changed=vendor/github.com/containerd/containerd/api/services/snapshots/v1/snapshots.proto
       >   cargo:rerun-if-changed=vendor/
       >
       >   --- stderr
       >   thread 'main' panicked at /build/tardev-snapshotter-3.2.0.azl1-vendor.tar.gz/prost-build/src/lib.rs:1457:10:
       >   Could not find `protoc` installation and this build crate cannot proceed without
       >       this knowledge. If `protoc` is installed and this crate had trouble finding
       >       it, you can set the `PROTOC` environment variable with the specific path to your
       >       installed `protoc` binary.If you're on debian, try `apt-get install protobuf-compiler` or download it from https://github.com/protocolbuffers/protobuf/releases
       >
       >   For more information: https://docs.rs/prost-build/#sourcing-protoc
```